app/src/main/java/com/dessalines/thumbkey/ui/components/keyboard/KeyboardKey.kt

```diff
@@ -363,7 +364,12 @@ fun KeyText(key: KeyC, keySize: Dp, hideLetters: Boolean, capsLock: Boolean) {
         }
         is KeyDisplay.TextDisplay -> {
             // Only  hide the letters for text, not symbols
-            if (!hideLetters) {
+            val showKey = if (!display.text.any { !it.isLetter() }) {
```

Comment:
What's going on here?

This isn't going to work for things like multiple letter keys, etc. You're going to need to add a `KeyType enum {Letter, Symbol}` to each key, probably with a default of `KeyType.Letter`, and alter every keyboard to set that.

Reply:
I'm using isLetter() on every character in the string. For example: "a" is a letter, "abc" is a letter, and "?" is a symbol, but "a?" is also a symbol. My assumption is that anything that is purely letter-characters is probably basically a letter.

Reply:
Okay, but I'd move the `isLetter` check above, and have this do a simple `if (!hideLetters || !isMadeOfLetters)`

Reply:
Because I made hiding symbols a separate toggle, I found that I have to use the `isLetter` check to determine which of `hideLetters` and `hideSymbols` to apply. I tried a bunch of logic like what you're describing in a python script and none of my attempts correctly resolved every possibility.

Reply:
So you haven't tested this code in thumbkey?

Reply:
I have tested it in the app. I have tested it with ThumbKeyENv4Symbols and ThumbKeyITv1. I wanted to test the `.com` in ThumbKeyENv4Programmer, but that key doesn't work for me, even on the official build.

Reply:
To clarify: I tested ThumbKeyITv1 because it was one of the only layouts that has multiple-letter keys ("qu" for example)

Reply:
Mmmk, I'll test this shortly.

Reply:
Nevermind, I apparently missed the fact that the ".com" input was only in the shifted layout. That's my bad, and also, I'm glad I figured that out, because it's pretty useful.

Reply:
Currently, `.com` is hidden with hideSymbols. I'm not sure if I necessarily intended that when I initially wrote it, but in a way it feels like the behaviour you would expect? Obviously, this is a particular edge case that could go either way, and it's a reason to potentially explicitly label each key as a symbol or not a symbol, but I think it's probably fine.

Reply:
Bad news: it _does_ cause issues with the Greek keyboard, because combining accents are not letters:
https://github.com/dessalines/thumb-key/blob/71348911eebd40bd5608a88f31cf9e6d85d32c11/app/src/main/java/com/dessalines/thumbkey/keyboards/ThumbKeyGRv1.kt#L294-L295

Reply:
Hrm... Could you check if there exists any other kotlin / java function would work? Might be something better than `isLetter`

Reply:
If not, even if imperfect, this would probably still be better than just hiding everything as it currently does.

Reply:
There's no better single function to my knowledge. It's possible to compare the character's Unicode category, and check if it's one of the 3 "mark" categories. I'm not 100% what all is in the mark categories overall, but punctuation and other special symbols generally shouldn't be.